### ISSUE TITLE
More efficient frontend image sending.

### ISSUE THREAD

**Issue by Krzysztofz01** [wailsapp/wails#2890]
### Is your feature request related to a problem? Please describe.

I am working on an application that generates images on the *"backend"* side and then sends them to the *"frontend"* as a preview.

From measurements, it seems that a lot of time is wasted on the very stage of preparing the data for transport and transporting them between the *"backend"* and the *"frontend"*. 
For context, the "heavy" image processing step takes approx. 300ms and the transport stage approx. +800ms.
This measurement was taken on Linux with the GNOME 44.4 DE and Wails version 2.5.1.

The current approach is to encode the result image to `base64`, send this string via a bind function to the *"frontend"* and set the `src` attribute of the `<img>` element.

### Describe the solution you'd like

Above all, I would like to ask those more experienced in using Wails if I have overlooked some simpler obvious solution.

I've been thinking about providing an API that allows you to submit primitive types bypassing the logic of mapping data to JavaScript/TypeScript structures.

I was also thinking about the possibility of streaming results using other network protocols such as WebSockets/WebRTC, but an issue dedicated to this already exists.

### Describe alternatives you've considered

If it turned out that the problem was caused, for example, by the very long `base64` interpretation time of the HTML engine. We could try to send a RGB byte array through the binding functions and iterate over a `<canvas>` element, which should work relatively efficiently thanks to the support for hardware acceleration. The problem is, however, that the amount of data (measured in bytes) to be sent to the *"frontend"* will be much larger.

Alternatively, we can try to send the image as a event payload, but this will require a lot of changes to the whole application, and the events themselves (according to what I read in a another issue), do not necessarily keep their order of emission.

**Comment by stffabi** [comment]
Thanks for using Wails and opening this request.

Sounds like this might be a good use case for an AssetServer request that generates the images. There's a tutorial which should give an example how this might be achieved [Dynamic Assets](https://wails.io/docs/guides/dynamic-assets). That should be much faster for transferring images.

**Comment by stffabi** [comment]
However, there is one limitation: Windows cannot currently process requests in parallel. I have a branch for that ready, hopefully I find time to open a PR soon.

**Comment by stffabi** [comment]
A PR for concurrent processing of AssetServer requests on Windows is now up #2926
